### ISSUE TITLE
NT_Interfaces_v2 can't convert a interface from DHCP to Static

### ISSUE THREAD

**Issue by dfduarte** [target/network_interfaces_v2-cookbook#13]
Hello There,

I'm using this cookbook to set some Windows interfaces here, in a project what I'm working. The problem is:

When I set the Provider to set a existent interface from dhcp to static, using the same configuration, the cookbooks bails out saying that the configuration its already up to date (may be, some guard parameter somewhere), and it cant change the interface configuration.

For example:

Interface IP: 192.168.1.89
mask: 255.255.252.0
gateway: 192.168.0.1
IP attribution format: DHCP

So, I do use the following:

network_interface 'Local Area Connection 3' do
  hw_address '08-00-27-7F-59-DB'
  bootproto 'static'
  address '192.168.1.89'
  netmask '255.255.252.0'
  gateway '192.168.0.1'
  dns ['192.168.0.1', '8.8.8.8']
end

When I use the above, is supposed that the cookbook should convert the interface to use a static address, with the same configuration provided by network. However, the cookbook finishes throwing a "already up to date" message.

Output:

Recipe: ac-windows-setup::default
  * win_network_interface[Ethernet] action create[2015-10-08T18:20:50+00:00] INFO: Processing win_network_interface[Ethernet] action create (ac-windows-setup::default line 77)
 (up to date)
[2015-10-08T18:20:51+00:00] INFO: Chef Run complete in 1.778392 seconds

The otherwise works perfectly, transforming the interface configuration in DHCP.

Maybe, the error consists in some logic error on IP address (or something like this) verification.

Edit - Additional notes: I guess that is a guard parameter logic error, because when I change the IP Address number, the conversion for "static" works flawlessly.

**Comment by jmccann** [comment]
I just want to make sure I'm understanding your situation correctly.

You have an interface that is currently configured for DHCP.  You write a block like the following (from your comment) to make it static:

```ruby
network_interface 'Local Area Connection 3' do
  hw_address '08-00-27-7F-59-DB'
  bootproto 'static'
  address '192.168.1.89'
  netmask '255.255.252.0'
  gateway '192.168.0.1'
  dns ['192.168.0.1', '8.8.8.8']
end
```

Chef says the resource is `up-to-date`, does NOTHING, and the interface is still DHCP.

Did I understand correctly?

**Comment by dfduarte** [comment]
Good morning @jmccann 

Yeay, that's right! I have a interface configured for DHCP, and I want to set these configurations from DHCP to a static manner. I did that, but the interface stills configured in DHCP mode, and nothing changes.

**Comment by jmccann** [comment]
Ok, perfect. I'll try to find time to take a look. Thanks.

**Comment by jmccann** [comment]
I believe I've addressed this and released 2.4.1(will release to supermarket soon).  Please give it a try.  If the issue persists feel free to reopen this.

**Comment by dfduarte** [comment]
Thanks A LOT @jmccann ! Worked like a charm!
